### ISSUE TITLE
Why covered_percent return 100% on no files or no lines covered?

### ISSUE THREAD

**Issue by wnuqui** [colszowka/simplecov#603]
This is more of a question (though I personally think this is a bug). Why even no code (no lines) written (just gem skeleton with testing setup) SimpleCov reports the following:

```bash
Coverage report generated for RSpec to /path/to/coverage. 0.0 / 0.0 LOC (100.0%) covered.
```

Just to make sure others can reproduce the in question behavior of SimpleCov, here are the steps to reproduce:

### Steps to reproduce
1. Generate a gem via bundle.
2. Use RSpec for testing.
3. Use SimpleCov for code coverage.
4. Then, in `spec_helper.rb` comment out `require "counter"` line

```ruby
require 'simplecov'
SimpleCov.start

require "bundler/setup"
#require "counter"

RSpec.configure do |config|
  # Enable flags like --only-failures and --next-failure
  config.example_status_persistence_file_path = ".rspec_status"

  config.expect_with :rspec do |c|
    c.syntax = :expect
  end
end
```

4. Run `bundle exec rspec`

### Current Behavior
- After running last step, you will see this text at the bottom:
```bash
Coverage report generated for RSpec to /path/to/coverage. 0.0 / 0.0 LOC (100.0%) covered.
```
- Even, `simplecov-html` is using the wrong percentage:
![100-percent-no-files-covered](https://user-images.githubusercontent.com/1292222/28993700-56dad876-79ef-11e7-9b01-b4dfceb73eed.png)

### Expected Behavior (in my opinion)
- It should not display `(100.0%) covered.`
- It should rather put a notification saying (for lack of better text):
```bash
Coverage report generated for RSpec to /path/to/coverage. 0.0 / 0.0 LOC covered. Did you correctly setup SimpleCov?
```
- In the reproduce steps, `counter` file is not required after `SimpleCov.start` so it will not be covered hence should aptly notify correct coverage numbers and say  something (in this case `counter` not required) is incorrectly configured/setup.
- `simplecov-html` should report `No Files covered` as header text instead.

### Related ruby and gem versions
```ruby
ruby 2.4.0p0 (2016-12-24 revision 57164) [i686-linux]

simplecov (0.14.1)
  docile (~> 1.1.0)
  json (>= 1.8, < 3)
  simplecov-html (~> 0.10.0)
```

### Investigation
I checked the code and I found that [`covered_percent`](https://github.com/colszowka/simplecov/blob/4f90f03a78be02c060ac899b9e750af8b8a9ba50/lib/simplecov/file_list.rb#L47-L50) is the related method which is defined as:
```ruby
# Computes the coverage based upon lines covered and lines missed
# @return [Float]
def covered_percent
  return 100.0 if empty? || lines_of_code.zero?
  Float(covered_lines * 100.0 / lines_of_code)
end
```

Why return `100.0` if `empty? || lines_of_code.zero?` is evaluated to `true`?

I'll be happy to create a PR to return `0.0` instead of `100.0` and as well fix `output_message` and html view in `simplecov-html` gem. cc @colszowka @PragTob

**Comment by PragTob** [comment]
:wave: 

thanks for your interest :) Imo it is not a bug although I had to think about it when I first came across it. Iirc it also applies to singular files. Think about files that are fully ignored or files that are just comments (aka no relevant lines) - should they show up as uncovered aka 0%? I don't think so, they would be marked as red in the html formatter which is like "this is an action item, go fix it!" but there is nothing to fix. Even more, let's say I got one 0/0 file and the rest of my code is 100%. Total coverage would show as 100% - which is correct, but then the one 0/0 file would show as 0% which is really confusing imo.

As for the case when the whole project is 0/0 - I agree it's more weird there and one could go either way on it. But I'd prefer it to be consistent with the single file instead of special casing it. I believe the 0.0/0.0 and total lack of any coverage data should be a big enough hint for anyone to see that something isn't setup correctly :)

On another note I agree though, if someone accidentally starts ignoring their whole code base (yay regexes!) the minimum_coverage option won't make it fail. But that is also difficult/can't solve everything - i.e. if just a file with little code is in there and covered - like 2 lines, then the rule won't apply although it's inherently the same problem.
